src/transformStateWithClones.js

```diff
@@ -7,7 +7,35 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateHistory = [];
+  const stateCopy = { ...state };
+
+  for (const obj of actions) {
+    switch (obj.type) {
+      case 'addProperties':
+        for (const key in obj.extraData) {
+          stateCopy[key] = obj.extraData[key];
+        }
+        stateHistory.push({ ...stateCopy });
```

Comment:
The `stateCopy` object is being mutated within the `switch` statement which contradicts the requirement to not mutate objects. Instead, create a new copy of the `stateCopy` object within each case before modifying it.

Reply:
that's a new object, I can change it